src/api/routes/myGroups.ts

```diff
@@ -45,41 +57,44 @@ const myGroups = router({
       return "/fakeMeeting";
     }
 
-    if (group.meetingLink && !meetingLinkIsExpired(group.updatedAt)) {
-      return group.meetingLink;
-    }
-
-    const now = Math.floor(Date.now() / 1000);
-    const groupName = formatGroupName(group.name, group.groupUsers.length);
-    const res = await createMeeting(encodeMeetingSubject(group.id, groupName), now, now + 3600);
-    invariant(res.meeting_info_list.length === 1);
-
-    const meetingLink = res.meeting_info_list[0].join_url;
-
-    await Group.update({
-      meetingLink,
-    }, {
-      where: {
-        id: group.id,
+      // if the user's group is present in OngoingMeetings Model, return the meeting link 
```

Comment:
Add a TODO to implement locking: https://stackoverflow.com/a/48297781. But don't do the actual implementation in this PR to keep things simple.